src/pymor/reductors/aaa.py

```diff
@@ -0,0 +1,417 @@
+                if rk > max_rk:
+                    max_rk = rk
+            max_rks.append(max_rk)
+        # exploit nullspace structure to obtain final max rank
+        denom = np.prod([len(self.itpl_part[k])-max_rks[k] for k in range(len(self.itpl_part))])
+        if denom == 0 or d_nsp % denom != 0:
+            return None
+        max_rks[max_idx] = len(self.itpl_part[max_idx]) - d_nsp / denom
+        max_rks[max_idx] = round(max_rks[max_idx])
+        for i in range(len(max_rks)):
+            self.itpl_part[i] = self.itpl_part[i][0:max_rks[i]+1]
+
+        # solve LS problem
+        L = full_nd_loewner(self.samples, self.sampling_values, self.itpl_part)
+        _, S, V = spla.svd(L, full_matrices=False, lapack_driver='gesvd')
+        VH = np.conj(V.T)
+        coefs = VH[:, -1:]
+
+        return coefs
+
+
+def nd_loewner(samples, svs, itpl_part):
+    """Compute higher-dimensional Loewner matrix using only LS partitions.
+
+    .. note::
+       For non-parametric data this is simply the regular Loewner matrix.
+
+    Parameters
+    ----------
+    samples
+        Tensor of samples (see :class:`pAAAReductor`).
+    svs
+        List of sampling values (see :class:`pAAAReductor`).
+    itpl_part
+        Nested list such that `itpl_part[i]` is a list of indices for interpolated
+        sampling values in `svs[i]`.
+
+    Returns
+    -------
+    L
+        (Parametric) Loewner matrix based only on LS partition.
+    """
+    d = len(samples.shape)
+    ls_part = [sorted(set(range(len(s))) - set(p)) for p, s in zip(itpl_part, svs)]
+
+    s0 = svs[0]
+    s = s0[itpl_part[0]]
+    sh = s0[ls_part[0]]
+    sd = sh[:, np.newaxis] - s
+    sdpd = sd
+    for i in range(1, d):
+        p0 = svs[i]
+        p = p0[itpl_part[i]]
+        ph = p0[ls_part[i]]
+        pd = ph[:, np.newaxis] - p
+        sdpd = np.kron(sdpd, pd)
+    samples0 = samples[np.ix_(*itpl_part)].reshape(-1)
+    samples1 = samples[np.ix_(*ls_part)].reshape(-1, 1)
+    samplesd = samples1 - samples0
+
+    return samplesd / sdpd
+
+
+def full_nd_loewner(samples, svs, itpl_part):
+    """Compute higher-dimensional Loewner matrix using all combinations of partitions.
+
+    .. note::
+       For non-parametric data this is simply the regular Loewner matrix.
+
+    Parameters
+    ----------
+    samples
+        Tensor of samples (see :class:`pAAAReductor`).
+    svs
+        List of sampling values (see :class:`pAAAReductor`).
+    itpl_part
+        Nested list such that `itpl_part[i]` is a list of indices for interpolated
+        sampling values in `svs[i]`.
+
+    Returns
+    -------
+    L
+        (Parametric) Loewner matrix based on all combinations of partitions.
+    """
+    L = nd_loewner(samples, svs, itpl_part)
+    range_S = range(len(svs))
+
+    # consider all combinations of variables coming from interpolation vs LS partition
+    for i in itertools.product(*([0, 1] for _ in range_S)):
```

Comment:
Which number (0 or 1) corresponds to interpolation vs LS partition?

Maybe something like this would be better?

```suggestion
    for svs_itpl in itertools.product(*([False, True] for _ in range_S)):
```